src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticService.cs

```diff
@@ -0,0 +1,118 @@
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeLocalFunctionStatic
+{
+    [Shared]
+    [ExportLanguageService(typeof(MakeLocalFunctionStaticService), LanguageNames.CSharp)]
+    internal sealed class MakeLocalFunctionStaticService : ILanguageService
+    {
+        private static readonly char[] s_underscore = { '_' };
+        private readonly SyntaxGenerator s_generator = CSharpSyntaxGenerator.Instance;
+
+        internal async Task<Document> CreateParameterSymbolAsync(Document document, LocalFunctionStatementSyntax localfunction, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(true);
+            var localFunctionSymbol = semanticModel.GetDeclaredSymbol(localfunction, cancellationToken);
+            var dataFlow = semanticModel.AnalyzeDataFlow(localfunction);
+            var captures = dataFlow.CapturedInside;
+
+            var parameters = CreateParameterSymbol(captures);
+
+            //Finds all the call sites of the local function
+            var workspace = document.Project.Solution.Workspace;
+            var arrayNode = await SymbolFinder.FindReferencesAsync(localFunctionSymbol, document.Project.Solution, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            var rootOne = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var editor = new SyntaxEditor(rootOne, s_generator);
+
+
+            foreach (var referenceSymbol in arrayNode)
+            {
+                foreach (var location in referenceSymbol.Locations)
+                {
+                    var root = await location.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                    var syntaxNode = root.FindNode(location.Location.SourceSpan); //Node for the identifier syntax
+
+
+                    var invocation = (syntaxNode as IdentifierNameSyntax).Parent as InvocationExpressionSyntax;
+                    if (invocation == null)
+                    {
+                        return document;
+                    }
+
+                    var argList = invocation.ArgumentList;
+                    List<ArgumentSyntax> x = new List<ArgumentSyntax>();
+
+                    foreach (var parameter in parameters)
+                    {
+
+                        var newArgument = GenerateArgument(parameter, parameter.Name, false);
+                        x.Add(newArgument as ArgumentSyntax);
+                    }
```

Comment:
this loop as it is now can be converted into a linq query like this (also please avoid using variable names like "x", try to be descriptive)

```cs
var newArguments = parameters.Select(p => GenerateArgument(p, p.Name, false) as ArgumentSyntax);
```